src/Symfony/Bridge/Monolog/Processor/RouteProcessor.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Monolog\Processor;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FinishRequestEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Adds the current route information to the log entry.
+ *
+ * @author Piotr Stankowski <git@trakos.pl>
+ */
+class RouteProcessor implements ProcessorInterface, EventSubscriberInterface
+{
+    private $routeData = array();
+    private $includeParams;
+
+    public function __construct(bool $includeParams = true)
+    {
+        $this->includeParams = $includeParams;
+    }
+
+    public function __invoke(array $records)
+    {
+        if (!empty($this->routeData) && !isset($records['extra']['requests'])) {
+            $records['extra']['requests'] = $this->routeData;
+        }
+
+        return $records;
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if ($event->isMasterRequest()) {
+            $this->routeData = array();
+        }
+
+        $request = $event->getRequest();
+        if (!$request->attributes->has('_controller')) {
+            return;
+        }
+
+        $currentRequestData = array(
+            'controller' => $request->attributes->get('_controller'),
+            'route' => $request->attributes->get('_route'),
+        );
+
+        if ($this->includeParams) {
+            $currentRequestData['route_params'] = $request->attributes->get('_route_params');
+        }
+
+        $this->routeData[] = $currentRequestData;
+    }
+
+    public function onKernelFinishRequest(FinishRequestEvent $event)
+    {
+        if (!$event->getRequest()->attributes->has('_controller')) {
+            return;
```

Comment:
This implementation seems a bit fragile. Here you are supposing nobody added a `_controller` attribute after `RouteProcessor::onKernelRequest`. But if somebody adds this attribute you will pop the wrong `routeData`.

Instead of `$this->routeData = []` I would use a `$this->routeData = new SplObjectStorage()`

Reply:
Very good point. I fixed it, but I had some problems with using `SplObjectStorage` - it's a bit unwieldy. For instance, I can't just use `array_values` on it. So instead, I've simply used `spl_object_id` to calculate array indexes. Is that OK?